tests/languages/download_test.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+
+import _thread as thread
+import platform
+from http.server import BaseHTTPRequestHandler
+from http.server import HTTPServer
+from pathlib import Path
+
+from pre_commit.languages import download
+from testing.language_helpers import run_language
+
+
+class MyRequestHandler(BaseHTTPRequestHandler):
+    def do_GET(self):
+        self.send_response(200)
+        self.end_headers()
+        self.wfile.write(b'#!/bin/sh\necho hello\nexit 123')
+
+
+class server():
+    def __init__(self):
+        self.httpd = HTTPServer(('localhost', 5555), MyRequestHandler)
+
+    def start(self):
+        thread.start_new_thread(self.httpd.serve_forever, ())
+
+    def stop(self):
+        self.httpd.shutdown()
+
+
+def test_download_dependencies(tmp_path: Path) -> None:
+    http = server()
+    http.start()
+    ret = run_language(
+        tmp_path,
+        download,
+        'test.sh',
+        deps=(
+            """linux/amd64
+sha256-oRJkj6Cr8nWIivZ9d3W+rVZt/aSW1l9YtxSVh+GtIHM=
+http://127.0.0.1:5555
+test.sh""",
+        ),
+    )
+    http.stop()
+    # We don't expect this to pass under Windows.
+    if platform.system() != 'Windows':
+        assert ret == (123, b'hello\n')
+    else:
+        assert True
```

Comment:
This should be quite straight forward to test on Windows, which seems sensible to do the effort.

We can return a valid `test.bat` from the web server, which is a simple as returning 

```py
self.wfile.write(b'echo hello\nexit /b 123')
```

As long as we save that as `test.bat`, we _should_ be able to run the `test` entry point on all platforms if we save `test.sh` as `test`.

Reply:
On modern Windows, you can just type `python` in a command terminal and it will redirect you to the Windows Store to install it.

Reply:
Without the ``#!/bin/bash``, it wouldn't run under pre-commit for some reason.

Reply:
The latest implementation works under both Windows and Linux. I have just manually tested it out.